### PR TITLE
Local completions

### PR DESCRIPTION
**Pull request by aviatesk**
Enables local completions in `julia_prompt`

### CODE REVIEW

File: src/repl.jl [hunk 1]
```diff
@@ -220,14 +220,28 @@ end
...
+    frame = c.state.frame
+
+    # repl backend completions
+    comps, range, should_complete = REPLCompletions.completions(full, lastindex(partial), moduleof(frame))
+    ret = map(REPLCompletions.completion_text, comps) |> unique!
+
+    # local completions
+    vars = filter!(locals(frame)) do v
+        # xref: https://github.com/JunoLab/Atom.jl/blob/master/src/debugger/stepper.jl#L411-L421
+        if v.name == Symbol("#self") && (v.value isa Type || sizeof(v.value) == 0)
```
Comment (aviatesk) [inline src/repl.jl]: we may remove the latter filter conditions here.

Comment (KristofferC) [reply src/repl.jl]: We have that in JuliaInterpreter as well but I am not sure I see the reason for it. I think it can be removed here anyway.

**Comment by KristofferC** [comment]
Thanks, this is a very nice enhancement.

File: src/repl.jl (hunk 1 above)
Comment (aviatesk) [reply src/repl.jl]: hm, maybe @timholy knows that ?

File: src/Debugger.jl [hunk 2]
```diff
@@ -27,7 +27,7 @@ end
 export @enter, @run
 
 # reexport some useful things from JuliaInterpreter
-using JuliaInterpreter: @bp, @breakpoint, breakpoint, break_on, break_off
+using JuliaInterpreter: @bp, @breakpoint, breakpoint, break_on, break_off, remove, disable
```
Comment (KristofferC) [inline src/Debugger.jl]: Can we take these for another PR... I don't really like exporting these super-generic function names (ref https://github.com/JuliaDebug/JuliaInterpreter.jl/issues/281)

File: src/repl.jl (hunk 1 above)
Comment (aviatesk) [reply src/repl.jl]: ops, sorry I found those functions aren't defined, thus I suspected you just forgot to import them.
Do you want to still export them ?

File: src/Debugger.jl (hunk 2 above)
Comment (aviatesk) [reply src/Debugger.jl]: ops, sorry I found those functions aren't defined, thus I suspected you just forgot to import them.
Do you want to still export them ?

Comment (aviatesk) [reply src/Debugger.jl]: Just reverted for now :)

Comment (KristofferC) [reply src/Debugger.jl]: Oh wait, this isn't exporting them. Then it's fine.

Comment (aviatesk) [reply src/Debugger.jl]: you didn't import them, but exported

Comment (KristofferC) [reply src/Debugger.jl]: Haha, okay!

File: src/repl.jl (hunk 1 above)
Comment (timholy) [reply src/repl.jl]: The `#self` is probably sufficient. It's not always there (I can't remember where I recently encountered such a situation), but it should be sufficient. Assuming I started that trend I was probably just being paranoid.

**Pull request merged**